app/assets/javascripts/pretty-text/engines/discourse-markdown/resize-controls.js

```diff
@@ -147,6 +79,36 @@ export function setup(helper) {
     ]);
 
     helper.registerPlugin(md => {
+      const oldRule = md.renderer.rules.image;
+
+      md.renderer.rules.image = function(tokens, idx, options, env, slf) {
+        var token = tokens[idx];
+        const scaleIndex = token.attrIndex("scale");
+        const imageIndex = token.attrIndex("index-image");
+
+        if (scaleIndex !== -1) {
+          var scale = token.attrs[scaleIndex][1];
+          var index = token.attrs[imageIndex][1];
+
+          let result = "<span class='image-wrapper'>";
+          result += oldRule(tokens, idx, options, env, slf);
+          result +=
+            "<span class='button-wrapper' data-image-index='" + index + "'>";
+
+          result += buildScaleButton(scale, "100");
```

Comment:
Then, using the constant above defined per the comment above you could replace this block with:

```suggestion
          result += PREDEFINED_SCALES
            .map(s => buildScaleButton(scale, s))
            .join("<span class='separator'>&nbsp;•&nbsp;</span>");
```

`PREDEFINED_SCALES` and `s` are not the best names, but I cannot think of better ones (maybe `s` could be `scale` and rename outer scope `scale` to `selectedScale`). :thinking: